server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -116,4 +116,23 @@ public static float updateNodeUtilizationWithShardMovements(
         float newNodeUtilization = nodeUtilization + (shardWriteLoadDelta / numberOfWriteThreads);
         return (float) Math.max(newNodeUtilization, 0.0);
     }
+
+    /**
+     * Update the max thread pool queue latency by accounting for the write load change due to shard movements
+     * @param maxThreadPoolQueueLatencyMillis The current max thread pool queue latency.
+     * @param shardWriteLoadDelta The write load change due to shard movements. Negative values indicate shards moving away.
+     *                            Positive values indicate shards moving onto the node.
+     * @return The new adjusted max thread pool queue latency.
+     */
+    public static long updateThreadPoolQueueLatencyWithShardMovements(long maxThreadPoolQueueLatencyMillis, float shardWriteLoadDelta) {
+        // Intentionally keep it simple by reducing queue latency to zero if we move any shard off the node.
+        // This means the node is no longer hot-spotting (with respect to queue latency) once a shard moves away.
+        // The next ClusterInfo will come in up-to 30 seconds later, and we will see the actual impact of the
+        // shard movement and repeat the process. We keep the queue latency unchanged if shard moves onto the node.
+        if (shardWriteLoadDelta < 0) {
```

Comment:
This seems to assume a shard write load exists and is != 0. I think we could instead simply track that the node had a shard move off it - and then return 0 queue latency if so, regardless of the write load?

Reply:
`replaceWritePoolStats` (which calls this) is only called if a one or more shards have been moved to or from the node in question during the simulation. A negative delta means we moved more load off than we moved on, but in reality probably just means we moved one or more shards away. I think this logic makes sense?

Reply:
I were thinking of the case where write load = 0, in which case we'd leave the queue latency in place. A write load of 0 seems possible?

Also, if there is a discrepancy between `writeLoadsPerShard` and the shard we end up moving (i.e., it is not in the map), we may also not clear the queue latency.

This seems like an unnecessary complication to think about when all we need is a boolean per node about whether the node did any relocations away from it?

Reply:
The idea was to go very simple initially. I dislike the idea of supporting a shard moving out and a similar one moving back in and then assuming the queue latency is still the same. I would prefer to make it simple for us to understand - if a shard left the node, then it takes another polling round for us to consider it hotspotted again.

It may even be the case that a shard moved to the node after the poll, then another shard left it. With this mechanism as is, we would then still consider it hotspotted and move more shards.

It all adds uncertainty on what to expect in prod, which is what I am trying to avoid.

Reply:
Thanks, look good.